src/System.CommandLine/Token.cs

```diff
@@ -15,6 +17,29 @@ internal Token(string value, TokenType type)
 
         public TokenType Type { get; }
 
+        public override bool Equals(object obj) =>
+            obj is Token token &&
+            Value == token.Value &&
+            Type == token.Type;
+
+        public override int GetHashCode()
+        {
+            var hashCode = 1574892647;
+            hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(Value);
+            hashCode = hashCode * -1521134295 + Type.GetHashCode();
+            return hashCode;
+        }
```

Comment:
This hashing function will not produce very randomly distributed hashes for `Tokens` with the same `Value`.
A few options:
```cs
public static int GetHashCode()
{
    unchecked
    {
        return ((int) Type * 397) ^ (Value != null ? Value.GetHashCode() : 0);
    }
}
```

Or leverage the [RyuJIT optimizations](https://source.dot.net/#System.Private.CoreLib/shared/System/Numerics/Hashing/HashHelpers.cs,12) that were done in `ValueTuple` 
```cs
public static int GetHashCode()
{
  return (Type, Value).GetHashCode();
}
```

Reply:
Fixing this in my other PR.